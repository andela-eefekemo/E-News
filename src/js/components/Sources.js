import React from 'react';
import { Link } from 'react-router';

const Sources = (props) => {
  const { name, description, id } = props;
  return (
    <div className="col m4">
      <h5>{name}</h5>
      <p>{description}</p>
      <Link class="btn btn-default" to={`/headlines:&{id}`} >More Info</Link>
    </div>
  );
};
export default Sources;
