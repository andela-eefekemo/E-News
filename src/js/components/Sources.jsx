import React from 'react';
import { browserHistory } from 'react-router';
import PropTypes from 'prop-types';

// Sources Component
class Sources extends React.Component {

  handleQueryValue(href) {
    browserHistory.push(href);
  }
  render() {
    const { name, description, id, sortBysAvailable } = this.props;
    const link = `/headlines?name=${name}&id=${id}&sorts=${sortBysAvailable}`;
    return (
      <div className="col s12 m6 l4 ">
        <h5>{name}</h5>
        <p className="truncate">{description}</p>
        <button
          className="btn btn-default"
          onClick={this.handleQueryValue.bind(this, link)}>
          View Headlines
        </button>
      </div>
    );
  }
}

// Set default Props
Sources.defaultProps = {
  name: '',
  description: '',
  id: '',
  sortBysAvailable: [],
};

Sources.propTypes = {
  name: PropTypes.string,
  description: PropTypes.string,
  id: PropTypes.string,
  sortBysAvailable: PropTypes.array,
};

export default Sources;
