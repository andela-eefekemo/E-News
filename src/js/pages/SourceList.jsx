import React from 'react';

import sourcesStore from '../stores/sourcesStore';
import * as NewsActions from '../actions/newsAction';
import Sources from '../components/Sources.jsx';

/**
 * Home React Component
 */
export default class SourceList extends React.Component {
  constructor() {
    super();
    this.state = {
      sources: [],
      name: 'sources',
      searchTerm: '',
    };
    this.sources = this.getSources.bind(this);
    this.filter = this.filterSources.bind(this);
  }
  /**
   * @method componentDidMount - Runs after the page has been rendered
   * @return {void}
   * Makes an action call to get list of sources from the Api
   * Listens for a change event from the sourcesStore
   */
  componentDidMount() {
    NewsActions.getSources();
    sourcesStore.on('changes', this.sources);
  }
  /**
   * @method componentWillUnmount - Runs after component has been closed
   * @return {void}
   * Removes changes Listener from the sourcesStore
   */
  componentWillUnmount() {
    sourcesStore.removeListener('changes', this.sources);
  }
  /**
   * @method getSources - Sets the state of Sources to data retrieve from sourcesStore
   * @return {void}
   */
  getSources() {
    this.setState({
      sources: sourcesStore.getSources(),
    });
  }
  /**
   * @method filterSources - Searches through available sources
   * @param {event} event - Takes in an onChange event
   * @return {void}
   * Maps through lists of articles retrieved from the sourcesStore
   * Sets the sources State to the search result
   */
  filterSources(event) {
    const searchList = [];
    sourcesStore.getSources().map((source) => {
      if (source.name.toLowerCase()
        .search(event.target.value.toLowerCase()) !== -1) {
        searchList.push(source);
      }
    });
    this.setState({
      sources: searchList,
      searchTerm: event.target.value,
    });
  }

  render() {
    const { sources } = this.state;
    // maps through sources array and passes props to Sources component
    const SourcesComponents = sources.map(source => <Sources key={source.id} {...source} />);
    return (
      <div className="container">
        <div className="row">
          <div className="col m8">
            <input
              id={this.state.name}
              type="text"
              className="validate"
              placeholder="Search for sources"
              onChange={this.filter} />
          </div>
        </div>
        <div className="row">
          <div>
            {SourcesComponents}
          </div>
        </div>
      </div>
    );
  }
}
