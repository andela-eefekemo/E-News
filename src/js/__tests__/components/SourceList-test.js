import React from 'react';
import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import SourceList from '../../pages/SourceList';
import Sources from '../../components/Sources';


test('Home Component', () => {
  const component = mount(<SourceList />);
  const tree = toJson(component);
  expect(tree).toMatchSnapshot();
  expect(component.find('#sources').length).toBe(1);
  expect(component.find('#bar').length).toBe(0);
  component.setState({ name: 'bar' });
  expect(component.find('#sources').length).toBe(0);
  expect(component.find('#bar').length).toBe(1);
});

test('Testing filter function', () =>{
  const filterSources = jest.fn();
  const component = mount(<SourceList onChange={filterSources} />);
  const input = component.find('input');
  expect(input).toBeDefined();
  expect(component.state().searchTerm).toBe('');
  input.simulate('change', {target: { value: 'the next' }});
  expect(component.state().searchTerm).toBe('the next');
});