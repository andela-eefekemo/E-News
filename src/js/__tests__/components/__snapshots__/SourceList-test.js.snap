// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Component 1`] = `
<SourceList>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="col m8"
      >
        <input
          className="validate"
          id="sources"
          onChange={[Function]}
          placeholder="Search for sources"
          type="text"
        />
      </div>
    </div>
    <div
      className="row"
    >
      <div />
    </div>
  </div>
</SourceList>
`;
